File: leetCode/fukun/101-200/lc_141.js
function ListNode(val) {
    this.val = val;
    this.next = null;
}

const hasCycle = function (head) {
    let curr = head;
    while (curr) {
        if (curr.flag) return true;
        curr.flag = true;
        curr = curr.next;
    }
    return false;
};

let ln0 = new ListNode(1);
let ln1 = new ListNode(2);
let ln2 = new ListNode(3);
ln0.next = ln1;
ln1.next = ln2;
console.log(hasCycle(ln0, 2, 2));

File: leetCode/linovce/lc_141.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode q = head;
        ListNode p = head;
        
        while(q!=null){
            q=q.next;
            if(q!=null)
                q=q.next;
            else
                return false;
            p=p.next;
            
            if(q==p)
                return true;
        }
        return false;
    }
}